### PR TITLE
[CoreBundle] support version marshall und unmarshall to merge with existing data

### PR DESCRIPTION
**Pull request by dpfaffenbauer**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  |no
| BC breaks?    | no
| Deprecations? |no
| Fixed tickets | #1118

### CODE REVIEW

File: src/CoreShop/Bundle/ResourceBundle/CoreExtension/Select.php [hunk 1]
```diff
@@ -33,6 +34,26 @@ abstract class Select extends Model\DataObject\ClassDefinition\Data\Select
...
+
+        return $data;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function unmarshalVersion($object, $data)
+    {
+        return $this->getRepository()->find($data);
```
Comment (solverat) [inline src/CoreShop/Bundle/ResourceBundle/CoreExtension/Select.php]: $data can be null

File: src/CoreShop/Bundle/ProductBundle/Resources/config/serializer/Model.ProductUnitDefinition.yml [hunk 2]
```diff
@@ -6,20 +6,26 @@ CoreShop\Component\Product\Model\ProductUnitDefinition:
...
             type: int
-            groups: [List, Detailed]
+            groups: [List, Detailed, Version]
         productUnitDefinitions:
             expose: true
             type: CoreShop\Component\Product\Model\ProductUnitDefinitions
-            groups: [List, Detailed]
\ No newline at end of file
+            groups: [List, Detailed]
+    virtual_properties:
```
Comment (solverat) [inline src/CoreShop/Bundle/ProductBundle/Resources/config/serializer/Model.ProductUnitDefinition.yml]: Its not possible to have two properties with the same name in serializer.

File: src/CoreShop/Bundle/ResourceBundle/CoreExtension/Select.php (hunk 1 above)
Comment (dpfaffenbauer) [reply src/CoreShop/Bundle/ResourceBundle/CoreExtension/Select.php]: solved

File: src/CoreShop/Bundle/ResourceBundle/Serialization/VersionObjectConstructor.php [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    public function construct(
+        VisitorInterface $visitor,
+        ClassMetadata $metadata,
+        $data,
+        array $type,
+        DeserializationContext $context
+    ) {
+        if ($context->getAttribute('unmarshalVersion')) {
```
[the remark this answers is not in the document]
Comment (dpfaffenbauer) [reply src/CoreShop/Bundle/ResourceBundle/Serialization/VersionObjectConstructor.php]: probably has and get right?

File: src/CoreShop/Bundle/ProductBundle/Resources/config/serializer/Model.ProductUnitDefinitions.yml [hunk 4]
```diff
@@ -6,16 +6,16 @@ CoreShop\Component\Product\Model\ProductUnitDefinitions:
...
             type: integer
             groups: [List, Detailed]
         defaultUnitDefinition:
             expose: true
             type: CoreShop\Component\Product\Model\ProductUnitDefinition
-            groups: [List, Detailed]
+            groups: [List, Detailed, Version]
         unitDefinitions:
             expose: true
             type: array<CoreShop\Component\Product\Model\ProductUnitDefinition>
```
[the remark this answers is not in the document]
Comment (dpfaffenbauer) [reply src/CoreShop/Bundle/ProductBundle/Resources/config/serializer/Model.ProductUnitDefinitions.yml]: not with our VersionObjectConstructor

**Pull request merged**